pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2316,6 +2316,7 @@ type SSSDSidecarAdditionalFile struct {
 
 // NetworkSpec for Ceph includes backward compatibility code
 // +kubebuilder:validation:XValidation:message="at least one network selector must be specified when using multus",rule="!has(self.provider) || (self.provider != 'multus' || (self.provider == 'multus' && size(self.selectors) > 0))"
+// +kubebuilder:validation:XValidation:message="the legacy hostNetwork setting can only be set if the network.provider is set to the empty string",rule="!has(self.hostNetwork) || !has(self.provider)"
```

Comment:
@parth-gr  wrote:
> rule="!has(self.hostNetwork) || !has(self.provider) -> rule="has(self.hostNetwork) && !has(self.provider)" might be this

Actually no: We want to prevent hostNetwork from being set together with a non-empty provider,  so the rule has to be this:
```
!(has(self.hostNetwork) && has(self.provider))
```
which Boolean algebra simplifies to:
```
!has(self.hostNetwork) || !has(self.provider)
```

Reply:
It is actually even more complex since we do need to check the values in addition to using `has()` I'll update shortly.